docs/env.md

```diff
@@ -1,51 +1,147 @@
-# Environment Variables
+## Passing Environment Variables to your application
 
-Sometimes it is practical to have different config values according to the environment that the application is running in.
+You application can run in different modes, as documented in the chapter about [Commands](commands.md) - `development`, `test`, `production`. Each of these modes might require different default values for things like:
 
-As an example:
+* log levels
+* API URLs
+* API access keys
+* ...?
 
-```js
-// config/prod.env.js
+You can define these values in the files with the name matching the mode.
+
+```javascript
+// config/variables/development.js
+module.exports = {
+  API_URL: 'dev.localhost:3000'
+}
+// config/variables/prod.env.js
+module.exports = {
+  API_URL: 'production.yourapp.com/api/v1'
+}
+```
+
+> If you are pulling your hair at this point because we are saving config in the code, please read on. We talk about those concerns in depth a bit further down.
+
+You can also map to environment variables easily.
+
+```javascript
+// config/variables/development.js
 module.exports = {
-  NODE_ENV: '"production"',
-  DEBUG_MODE: false,
-  API_KEY: '"..."' // this is shared between all environments
+  API_URL: process.env.API_URL
 }
+```
+
+You will also find a file called `externals.js` in this folder. Use this file if to map any environment variables that you want to pass to your Vue app independent of the mode.
```

Comment:
What is the reasoning behind the name "externals"?

Since this file is loaded regardless of "mode" it seems like the place to store the default values that will get overwritten with mode/env specific values - hence maybe "defaults" would make more sense than "externals"?